vowpalwabbit/options_types.h

```diff
@@ -30,9 +30,9 @@ struct typelist
 
 // is_scalar detection relies on only accepting std::vector<T> for vector types. If that changes, update
 // is_scalar_impl below.
-using supported_options_types =
-    typelist<unsigned int, int, size_t, uint64_t, float, double, char, std::string, bool, std::vector<int>,
-        std::vector<size_t>, std::vector<float>, std::vector<double>, std::vector<char>, std::vector<std::string>>;
+using supported_options_types = typelist<unsigned int, int, size_t, uint64_t, int64_t, float, double, char, std::string,
```

Comment:
what is the purpose of adding these 3 new types? or why would be a better question

Reply:
oh, I just noticed the other prs that depend on this one

Reply:
Yep, the plan is to constrain options to only a few essential types compared to the several duplicate types there are now